lightnion/path_selection.py

```diff
@@ -120,6 +156,42 @@ def keep_exit(router, state):
     return True, state, nhop
 
 
+def keep_exit_with_descr(descr, router, guard):
+    """Checks that the router is not a bad exit, is not down, is stable,
+    is valid, does not run an old TOR's version, has an available ed25519
+    identity key and has an 'accept' exit policy
+    :param descr: descriptor of the exit candidate.
+    :params router: the router we want to check
+    :param guard: the guard in the path
+    :return: tuple (boolean that indicates if we keep it, new state, descriptor)"""
+
+    if not obey_minimal_constraint(router, guard=guard):
+        return False, None
+
+    if 'Exit' not in router['flags']:
+        return False, None
+
+    if 'BadExit' in router['flags']:
+        return False, None
+
+    nhop = descr[router['digest']]
+
+    if router['digest'] != nhop['digest']:
+        return False, None
+
+    if 'identity' not in nhop or nhop['identity']['type'] != 'ed25519':
+        return False, None
```

Comment:
You're right, we don't have support for `EXTEND` cells towards nodes without `ed25519` keys.

It's clearly a tweak on our side, it can be a nice thing to add a comment here to point this out.